Sources/UI/NotificationCenter/NotificationCenterViewController.swift

```diff
@@ -23,75 +25,37 @@ open class NotificationCenterViewController: UIViewController {
     public private(set) var refreshControl = UIRefreshControl()
     public private(set) var tableView = UITableView()
     
-    private var cache: [Notification]?
-    private var notificationsObservation: NSObjectProtocol?
     private var didBecomeActiveObserver: NSObjectProtocol?
     
-    public var notifications: [Notification] {
-        if let cache = cache {
-            return cache
-        }
-        
-        let notifications = filterNotifications()
-        cache = notifications
-        return notifications
-    }
-    
-    /**
-     Returns a filtered list of notifications from dataStore.notifications. The default implementation filters out notifications that aren't notification center enabled, are deleted or have expired. This method is called automatically by the NotificationCenterTableView and the results are cached for performance.
-     
-     You can override this method if you wish to modify the rules used to filter notifications. For example if you wish to include expired notifications in the table view and instead show their expired status with a visual indicator.
-     */
-    open func filterNotifications() -> [Notification] {
-        return notificationStore.notifications.filter({ notification in
-            guard notification.isNotificationCenterEnabled, !notification.isDeleted else {
-                return false
-            }
-            
-            if let expiresAt = notification.expiresAt {
-                return expiresAt > Date()
-            }
-            
-            return true
-        })
-    }
-    
     public init(
         eventPipeline: EventPipeline,
-        imageStore: ImageStore,
-        notificationStore: NotificationStore,
         router: Router,
+        imageStore: ImageStore,
         sessionController: SessionController,
         syncCoordinator: SyncCoordinator,
+        managedObjectContext: NSManagedObjectContext,
         websiteViewControllerProvider: @escaping WebsiteViewControllerProvider) {
         self.eventPipeline = eventPipeline
-        self.imageStore = imageStore
-        self.notificationStore = notificationStore
         self.router = router
+        self.imageStore = imageStore
         self.sessionController = sessionController
         self.syncCoordinator = syncCoordinator
+        self.managedObjectContext = managedObjectContext
         self.websiteViewControllerProvider = websiteViewControllerProvider
         
         super.init(nibName: nil, bundle: nil)
         
-        notificationsObservation = notificationStore.addObserver { [weak self] _ in
-            DispatchQueue.main.async {
-                self?.cache = nil
-                self?.tableView.reloadData()
-            }
+        do {
```

Comment:
I don't know if this makes sense here. Do you need to fetch results in the initializer? This probably makes more sense in the `viewDidLoad` method. At this point you haven't even assigned the controller as the delegate to the fetchedResultsController so the result of this fetch would not even be used.